testcase.py

```diff
@@ -1054,6 +1062,100 @@ def get_version(self):
 
         return status, message, payload
 
+    def post_results_to_dashboard(self, api_key, tags):
+        '''Posts test results to online dashboard at given server address.
+
+        Parameters
+        ----------
+        api_key : str
+            API key corresponding to user account for dashboard.
+        tags : list of str
+            List of tags to be included with result posting.
+
+        Returns
+        -------
+        status: int
+            Indicates whether a posting to the dashboard has been successfully completed.
+            If 200, the posting was successful.
+            If 400, there was an input error.
+            If 500, an internal error occurred.
+        message: str
+            Includes detailed debugging information
+        payload: None
+
+        '''
+
+        # Check formal scnenario test completed
+        if not self.scenario_end:
+            status = 500
+            message = 'Formal test scenario, including time period, has not been completed.  Initialize a test scenario using the PUT /scenario API and run it to completion before submitting results to dashboard.'
+            logging.error(message)
+            return status, message, None
+        # Check parameters
+        if not isinstance(api_key, str):
+            status = 400
+            message = 'Invalid type for input api_key. It must be a string, but is a {0}.'.format(type(api_key))
+            logging.error(message)
+            return status, message, None
+
+        if not isinstance(tags, list):
+            status = 400
+            message = 'Invalid type for input tags. It must be a list of strings, but is a {0}.'.format(type(tags))
+            logging.error(message)
+            return status, message, None
+
+        if len(tags)>10:
+            status = 400
+            message = 'Invalid number of tags. The limit is 10, but there are {0}.'.format(len(tags))
+            logging.error(message)
+            return status, message, None
+
+        for tag in tags:
+            if not isinstance(tag, str):
+                status = 400
+                message = 'Invalid type for one of the tag inputs. They must be strings, but one is a {0}.'.format(type(tag))
+                logging.error(message)
+                return status, message, None
+        # Specify server address and payload
+        dash_server = os.environ['BOPTEST_DASHBOARD_SERVER']
+        payload = {
+          "results": [
+            {
+              "uid": str(int(random.random()*10000000)),
```

Comment:
A real uuid might be better. Like this?

https://docs.python.org/3/library/uuid.html

Reply:
Ah great, didn't know about that.  I updated in the following commit to use uuid4: https://github.com/ibpsa/project1-boptest/pull/444/commits/bcbbf29b049a8332a3907e9de6f06b11d26c07aa.